cellprofiler/modules/measureradialentropy.py

```diff
@@ -0,0 +1,187 @@
+
+        self.input_image_name = cps.ImageNameSubscriber(
+            "Select an image to measure", cps.NONE, doc="""Select the
+            grayscale image you want to measure the entropy of.""" )
+
+        self.bin_number=cps.Integer(
+            "Input number of bins", 6, minval=3, maxval=60,
+            doc="""Number of radial bins to divide your object into.  The minimum number
+            of bins allowed is 3, the maximum number is 60.""")
+
+        self.intensity_measurement=cps.Choice(
+            "Which intensity measurement should be used?", ['Mean','Median','Integrated'], value='Mean',doc="""
+            Whether each wedge's mean, median, or integrated intensity
+            should be used to calculate the entropy.""" )
+
+    def settings(self):
+        return [self.input_image_name, self.input_object_name,
+                self.intensity_measurement, self.bin_number]
+
+    def run(self, workspace):
+        measurements = workspace.measurements
+
+        statistics = [["Entropy"]]
+
+        workspace.display_data.statistics = statistics
+
+        input_image_name = self.input_image_name.value
+        input_object_name = self.input_object_name.value
+        metric = self.intensity_measurement.value
+        bins = self.bin_number.value
+
+        image_set = workspace.image_set
+
+        input_image = image_set.get_image(input_image_name,
+                                          must_be_grayscale=True)
+        pixels = input_image.pixel_data
+
+        object_set = workspace.object_set
+
+        objects = object_set.get_objects(input_object_name)
+        labels = objects.segmented
+
+        indexes = objects.indices
+        #Calculate the center of the objects- I'm guessing there's a better way to do this but this was already here
+        centers, radius = minimum_enclosing_circle(labels, indexes)
+
+        feature = self.get_measurement_name(input_image_name,metric,bins)
+        #Do the actual calculation
+        entropy=self.slice_and_measure_intensity(pixels,labels,indexes,centers,metric,bins)
+        #Add the measurement back into the workspace
+        measurements.add_measurement(input_object_name,feature,entropy)
+
+        emean = numpy.mean(entropy)
+        statistics.append([feature, emean])
+
+
+    ################################
+    #
+    # DISPLAY
+    #
+    def display(self, workspace, figure):
+        statistics = workspace.display_data.statistics
+        figure = workspace.create_or_find_figure(subplots=(1, 1,))
+        figure.subplot_table(0, 0, statistics)
+
+
+    def slice_and_measure_intensity(self, pixels, labels, indexes, centers, metric, nbins):
+        '''For each object, iterate over the pixels that make up the object, assign them to a bin,
+        then call calculate_entropy and return it to run.  Needs an update to numpy vector operations'''
+        entropylist=[]
+        for eachindex in range(len(indexes)):
```

Comment:
Since `indexes` and `centers` are lists of the same length you can process them as:

```python
for index, center in zip(indexes, centers):
    # do stuff
```

`indexes[eachindex]` becomes `index` and `centers[eachindex]` becomes `center`

You can break this up further as:

```python
for index, (center_y, center_x) in zip(indexes, centers):
    # do stuff
```

to remove the need for the line `center_y, center_x = center`